test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/TaskListControllerSpec.scala

```diff
@@ -82,15 +89,120 @@ class TaskListControllerSpec
 
-  }
+          inSequence {
+            mockAuthWithNoRetrievals()
+            mockGetSession(
+              Future.successful(
+                Right(
+                  Some(
+                    SessionData.empty.copy(
+                      journeyStatus = Some(fillingOutReturn)
+                    )
+                  )
+                )
+              )
+            )
+          }
+
+          val result = performAction()
+          status(result) shouldBe OK
+
+          val doc: Document = parse(contentAsString(result))
+          doc.title shouldBe messagefromMessageKey("service.title") + " - GOV.UK"
+
+          doc.select("li#propertyAddress > a").text shouldBe messagefromMessageKey("task-list.enter-property-address.link")
+          doc.select("li#propertyAddress > a").attr("href") shouldBe address.routes.PropertyAddressController.enterPostcode().url
+          doc.select("li#propertyAddress > strong").text shouldBe TaskListStatus.ToDo.toString
+        }
+
+        "the session data indicates that they are filling in a return and enter property address is complete" in {
+          val fillingOutReturn = sample[FillingOutReturn].copy(draftReturn = sample[DraftReturn]
+            .copy(propertyAddress = Some(sample[UkAddress])))
+
+          inSequence {
+            mockAuthWithNoRetrievals()
+            mockGetSession(
+              Future.successful(
+                Right(
+                  Some(
+                    SessionData.empty.copy(
+                      journeyStatus = Some(fillingOutReturn)
+                    )
+                  )
+                )
+              )
+            )
+          }
+
+          val result = performAction()
+          status(result) shouldBe OK
+
+          val doc: Document = parse(contentAsString(result))
+          doc.title shouldBe "Report and pay Capital Gains Tax on UK property - GOV.UK"
+
+          doc.select("li#propertyAddress > a").text shouldBe messagefromMessageKey("task-list.enter-property-address.link")
+          doc.select("li#propertyAddress > a").attr("href") shouldBe address.routes.PropertyAddressController.checkYourAnswers().url
+          doc.select("li#propertyAddress > strong").text shouldBe TaskListStatus.Complete.toString
+
+        }
+      }
+
+      "display the page with Save and come back later link" when {
+        "the session data indicates that they are filling in a return" in {
+          inSequence {
+            mockAuthWithNoRetrievals()
+            mockGetSession(
+              Future.successful(
+                Right(
+                  Some(
+                    SessionData.empty.copy(
+                      journeyStatus = Some(sample[FillingOutReturn])
+                    )
+                  )
+                )
+              )
+            )
+          }
+
+          val result = performAction()
+          status(result) shouldBe OK
 
+          val doc: Document = parse(contentAsString(result))
```

Comment:
could we assert the page title as well? It gives us a bit more confidence that we're on the right page

Reply:
I will discuss it personally with you, don't think we need nor want to check that.

Reply:
done